src/snowflake/cli/api/connections.py

```diff
@@ -80,6 +80,7 @@ def update(self, **updates):
             setattr(self, key, value)
 
     def update_from_config(self) -> ConnectionContext:
+        self.validate_and_complete()
```

Comment:
Sending telemetry data before execution was causing creation of connection, including connection name resolution (for example falling back to default conn). This method is called first here because it resolves `self.connection_name`. Without calling it the value is `None`